refactor(linter): gate rule docs behind feature

camchenry opened the pull request:
- related to https://github.com/oxc-project/oxc/issues/9998

Adds a `ruledocs` feature for capturing the documentation text in lint rule macros. This is only enabled for the website task currently.

Locally, this removes ~429K bytes from the binary size when running `cargo build --release -p oxlint`

<img width="247" alt="Screenshot 2025-03-25 at 12 18 23 AM" src="https://github.com/user-attachments/assets/52bbf254-eebf-45c5-8229-282fee921101" />

camchenry commented:
* **#10027** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10027?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10027?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10027).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
NICE!

Boshen commented:
### Merge activity

* **Mar 25, 12:50 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
refactor(linter): gate rule docs behind feature (#10027)

- related to https://github.com/oxc-project/oxc/issues/9998

Adds a `ruledocs` feature for capturing the documentation text in lint rule macros. This is only enabled for the website task currently.

Locally, this removes ~429K bytes from the binary size when running `cargo build --release -p oxlint`

<img width="247" alt="Screenshot 2025-03-25 at 12 18 23 AM" src="https://github.com/user-attachments/assets/52bbf254-eebf-45c5-8229-282fee921101" />
